tests/helpers/perf_results_helper.rb

```diff
@@ -603,8 +616,33 @@ def percent_diff_string(result_a, result_b)
   # @example
   #   extract_puppet_metrics_collector_data(metrics_dir)
   #
-  def extract_puppet_metrics_collector_data(metrics_dir)
-    raise "Directory not found: #{metrics_dir}" unless File.directory?(metrics_dir)
+  def extract_puppet_metrics_collector_data(metrics_dir_or_tar_file)
+    if !File.directory?(metrics_dir_or_tar_file)
+
+      extension = File.extname(metrics_dir_or_tar_file)
+      error_msg = "Specified path must be either a directory or tar file: #{metrics_dir_or_tar_file}"
+      raise error_msg unless extension == ".gz"
+
+      # change to the directory
+      original_dir = Dir.pwd
+      tar_file_dir = File.dirname(metrics_dir_or_tar_file)
+      Dir.chdir tar_file_dir
+
+      # extract
+      puts "Extracting tar file: #{metrics_dir_or_tar_file}"
+      command = "tar xfz #{File.basename(metrics_dir_or_tar_file)}"
+      `#{command}`
```

Comment:
Can minitar be used for this rather than exec'ing bash commands?

Reply:
Updated